azurerm/internal/services/desktopvirtualization/virtual_desktop_application_group.go

```diff
@@ -0,0 +1,191 @@
+			return tf.ImportAsExistsError("azurerm_virtual_host_pool", *existing.ID)
+		}
+	}
+
+	location := azure.NormalizeLocation(d.Get("location").(string))
+	t := d.Get("tags").(map[string]interface{})
+
+	ApplicationGroupType := d.Get("type").(string)
+	friendlyName := d.Get("friendly_name").(string)
+	description := d.Get("description").(string)
+	hostPoolID := d.Get("host_pool_id").(string)
+
+	context := desktopvirtualization.ApplicationGroup{
+		Location: &location,
+		Tags:     tags.Expand(t),
+		ApplicationGroupProperties: &desktopvirtualization.ApplicationGroupProperties{
+			ApplicationGroupType: desktopvirtualization.ApplicationGroupType(ApplicationGroupType),
+			FriendlyName:         &friendlyName,
+			Description:          &description,
+			HostPoolArmPath:      &hostPoolID,
+		},
+	}
+
+	_, err := client.CreateOrUpdate(ctx, resourceGroup, name, context)
+	if err != nil {
+		return fmt.Errorf("Error creating Virtual Desktop Host Pool %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	result, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		return fmt.Errorf("Error retrieving Virtual Desktop Host Pool %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	if result.ID == nil {
+		return fmt.Errorf("Cannot read Virtual Desktop Host Pool %q (Resource Group %q) ID", name, resourceGroup)
+	}
+	d.SetId(*result.ID)
+
+	return resourceArmVirtualDesktopApplicationGroupRead(d, meta)
+}
+
+func resourceArmVirtualDesktopApplicationGroupRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).DesktopVirtualization.ApplicationGroupsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.VirtualDesktopApplicationGroupID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[DEBUG] Virtual Desktop Host Pool %q was not found in Resource Group %q - removing from state!", id.Name, id.ResourceGroup)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error making Read request on Virtual Desktop Host Pool %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+	}
+
+	d.Set("name", id.Name)
+	d.Set("resource_group_name", id.ResourceGroup)
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+
+	return tags.FlattenAndSet(d, resp.Tags)
+}
+
+func resourceArmVirtualDesktopApplicationGroupDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).DesktopVirtualization.ApplicationGroupsClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.VirtualDesktopApplicationGroupID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	_, err = client.Delete(ctx, id.ResourceGroup, id.Name)
+	if err != nil {
```

Comment:
could we combine these two lines?
```suggestion
	if _, err = client.Delete(ctx, id.ResourceGroup, id.Name); err != nil {
```